sim/firesim-lib/src/main/cc/bridges/dromajo.cc

```diff
@@ -216,15 +217,19 @@ int dromajo_t::invoke_dromajo(uint8_t* buf) {
 /**
  * Read queue and co-simulate
  */
-void dromajo_t::process_tokens(int num_beats) {
+size_t dromajo_t::process_tokens(int num_beats, size_t minimum_batch_beats) {
+    auto beat_bytes = DMA_DATA_BITS / 8;
+    size_t maximum_batch_bytes = beats * beat_bytes;
+    size_t minimum_batch_bytes = minimum_batch_beats * beat_bytes;
     // TODO: as opt can mmap file and just load directly into it.
-    alignas(4096) uint8_t OUTBUF[QUEUE_DEPTH * sizeof(uint64_t) * 8];
-    pull(this->_dma_addr, (char*)OUTBUF, num_beats * sizeof(uint64_t) * 8);
+    alignas(4096) uint8_t OUTBUF[maximum_batch_bytes];
```

Comment:
> Is this `4096` here because you're assuming the underlying host uses 4KB pages and you want these to be on page boundaries? Oh, maybe this is just a requirement that comes from Dromajo (after looking at the comment on new line 233 below)? Actually, I see the answer in https://github.com/firesim/firesim/pull/1017/files#diff-a35f46844b5037a6aa1023bf9bb121c0ff85d0afa4e2e437d676c54794067d33R218-R222 . This is one instance where having a firesim-wide macro that is something like `PAGE_ALIGNED_BUFFER(OUTBUF, maximum_batch_bytes)`. 

Makes sense. 

> What happens if the page size isn't 4K?

I'm not sure. I'm assuming you're specifically talking about the behavior under super pages? 

> Also, why do you define the type as `uint8_t` and then cast it to `char *` in the call to `pull`? Or, is this one of the typing things you have listed to clean up in your TODO list?

I will clean this up.

Reply:
> I'm assuming you're specifically talking about the behavior under super pages?

Yeah, that's one definite case and I'm just wondering if we aught to (in the future) have some kind of runtime sanity test for anything like this.  There's a hardcoded assumption in the code that pages are 4K and if someone happened to run on a system where that was not the case, it would be good to have the driver blow up early and complain.  Perhaps capture this as an issue for future work?